WordPress/create-block-theme
# Bundle template images into theme assets and make their urls relative

Initially discussed on #192, continuing here for a cleaner git history.

## What?
If a template or a template part includes images which source is an absolute URL:
- The images are downloaded and bundled in the `/assets/images` folder of the theme
- The templates or template parts that include these images are converted to a pattern and added to the pattern folder. The template content is replaced by a reference to the pattern recently created.

Currently, we are considering the media content of these blocks:
- Image
- Cover
- Media & text
- Video

## Why?
To make the process of launching a theme easier by bundling to theme assets the images inserted using the editor.

## To Do:
Include other blocks including images as cover.

## How to test:
- Include one or more image blocks in a template or template part using the editor.
- Save the changes
- Export the theme using the plugin
- See if the resulting theme works as expected


Closes: #161
